saviSimulation/savi/jason_processing/SAVIWorld_model.java

```diff
@@ -105,82 +108,93 @@ public void setup() {
+		NUMBER_UGV = Integer.parseInt(modelProps.getProperty("NUMBER_UGV"));
+		NUMBER_UAV = Integer.parseInt(modelProps.getProperty("NUMBER_UAV"));
+		UGV_SIZE = Integer.parseInt(modelProps.getProperty("UGV_SIZE"));
+		UAV_SIZE = Integer.parseInt(modelProps.getProperty("UAV_SIZE"));
 
 		// Initialization code goes here
 		simTime = 0; // seconds
 		simTimeDelta = 1000 / FRAME_RATE; // milliseconds
-
 		simPaused = false;// not paused by default
 
+
 		playButton = new Button("play", width / 2 - 20, 10, 40, 40);
 		stopButton = new Button("restart", width / 2 + 20, 10, 40, 40);
 
 		play = loadShape("SimImages/play.svg");
 		pause = loadShape("SimImages/pause.svg");
-		restart = loadShape("SimImages/replay.svg");
 
 		// load images for visualization
+		restart = loadShape("SimImages/replay.svg");
 		treeImage = loadShape("SimImages/tree.svg");
 		houseImage = loadShape("SimImages/home.svg");
 		play = loadShape("SimImages/play.svg");
 		pause = loadShape("SimImages/pause.svg");
 		restart = loadShape("SimImages/replay.svg");
 		threatImage = loadShape("SimImages/warning.svg");
-		uasImage = loadShape("SimImages/robot.svg");
-		antennaImage = loadShape("SimImages/antenna.svg"); //TODO: get an antenna image
+		ugvImage = loadShape("SimImages/robot.svg");
+		uavImage = loadShape("SimImages/airplane.svg");
+		antennaImage = loadShape("SimImages/antenna.svg");
 
 		Random rand = new Random();
-
-		for (int i = 0; i < NUMBER_UAS; i++) { // Put UAS
-			// _PIXELS is the maximum and the 1 is our minimum
-			// TODO: right now agents are initialized with strings "0", "1", "2", ... as
-			// identifiers and a fixed type "demo" which matches their asl file name. This
-			// should be configurable...
-			if (RANDOM_SEED != -1) {
-				rand = new Random(RANDOM_SEED + i);
+		
+				
+		for(int i = 0; i < NUMBER_UAV+NUMBER_UGV; i++)  { //Put UxV
+			//_PIXELS is the maximum and the 1 is our minimum
+			//TODO: right now agents are initialized with strings "0", "1", "2", ... as identifiers and a fixed type "demo" which matches their asl file name. This should be configurable...
+			if(RANDOM_SEED != -1) {
+				rand = new Random(RANDOM_SEED+i);
+			}	
+		
+			if(i < NUMBER_UGV)  { //Put UgV
+				//_PIXELS is the maximum and the 1 is our minimum
+				//TODO: right now agents are initialized with strings "0", "1", "2", ... as identifiers and a fixed type "demo" which matches their asl file name. This should be configurable...
+				UgV ugv= new UgV(i, new PVector(rand.nextInt(X_PIXELS) + 1, rand.nextInt(Y_PIXELS) + 1, UGV_SIZE/2), UGV_SIZE/2,"demo", this, ugvImage, REASONING_CYCLE_PERIOD, "robot");
+				wifiParticipants.add(ugv.getAntennaRef());
+				objects.add(ugv);
+			}else {
+				UaV uav = new UaV(i, new PVector(rand.nextInt(X_PIXELS) + 1, rand.nextInt(Y_PIXELS) + 1, UAV_SIZE/2), UAV_SIZE/2,"demo", this, uavImage, REASONING_CYCLE_PERIOD, "airplane");
+				wifiParticipants.add(uav.getAntennaRef());
+				objects.add(uav);
 			}
```

Comment:
I think this is more readable if we just have one type of object per loop, as for threats, trees, etc.

Reply:
Agreed

Reply:
Fixed by @cruizm